tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/config/ConfigView.java

```diff
@@ -79,6 +81,13 @@ public void render(Map model, HttpServletRequest request, HttpServletResponse re
 
 				if (!headersOnly) {
 					Model configuration = (Model) model.get(CONFIG_DATA_KEY);
+					// obfuscate the password, if present
+					configuration.getStatements(null, HTTPRepositorySchema.PASSWORD, null).forEach(st -> {
+						configuration.add(st.getSubject(), HTTPRepositorySchema.PASSWORD,
+								SimpleValueFactory.getInstance().createLiteral("****"));
+						configuration.remove(st);
+					});
```

Comment:
A massive problem with this approach is that if a client is unaware credentials have been configured, and tries to modify some other configuration property by retrieving config from the server, changing one properrty, then sending back, will overwrite the existing password with one containing four asterisks.